packages/xo-server/src/xapi/index.mjs

```diff
@@ -1523,4 +1525,45 @@ export default class Xapi extends XapiBase {
       }
     }
   }
+
+  async getBiosInfo(ref, { cache } = {}) {
+    const biosData = await this.call('host.get_bios_strings', ref)
+
+    const { 'bios-version': currentBiosVersion, 'system-product-name': hostServerName } = biosData
+
+    if (biosData['system-manufacturer']?.toLowerCase() !== '2crsi') {
+      return
+    }
+
+    // this code has race conditions which might lead to multiple fetches in parallel
+    // but it's no big deal
+    let servers
+    if (!cache?.has('servers')) {
+      const response = await fetch(
+        'https://pictures.2cr.si/Images_site_web_Odoo/Pages_produit/VATES-BIOS_BMC_last-version.json'
+      )
+      const json = await response.json()
+      servers = keyBy(json[0]['2CRSi_Servers'], 'Server_Name')
+
+      cache?.set('servers', servers)
+    } else {
+      servers = cache.get('servers')
+    }
+
+    const serverData = servers?.[hostServerName]
+
+    if (serverData === undefined) {
+      return
+    }
+
+    const { 'BIOS-Version': latestBiosVersion, 'BIOS-link': biosLink } = serverData
+
+    function convertToSemver(version) {
+      return version.split('.').map(Number).join('.')
+    }
```

Comment:
- Rename the function to better match what it does. (`sanitizeVersion`/`normalizeVersion`)? cc/ @pdonias
- Place the function outside to avoid regenerating it every time `getBiosInfo` is called
- Add a comment to explain why this operation is required

Reply:
1. You can use `semver.clean(version, { loose: true })`.

Reply:
Or even simply `semver.eq(version1, version2, { loose: true })` to compare the versions.